src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/jvm/ThreadList.java

```diff
@@ -236,17 +275,24 @@ private static void parseThreadInfo(final ThreadInfo info) {
   }
 
   static void runThreadDump(String pid, String[] args) {
-    jTidNameMap.clear();
-    oldNativeTidMap.putAll(nativeTidMap);
-    nativeTidMap.clear();
-    jTidMap.clear();
-    nameMap.clear();
-
-    // TODO: make this map update atomic
-    Util.invokePrivileged(() -> runAttachDump(pid, args));
-    runMXDump();
-
-    lastRunTime = System.currentTimeMillis();
+    String currentThreadName = Thread.currentThread().getName();
+    if (currentThreadName.startsWith(
+        ScheduledMetricCollectorsExecutor.COLLECTOR_THREAD_POOL_NAME) ||
+        currentThreadName.equals(ScheduledMetricCollectorsExecutor.class.getSimpleName())) {
+      jTidNameMap.clear();
+      oldNativeTidMap.putAll(nativeTidMap);
+      nativeTidMap.clear();
+      jTidMap.clear();
+      nameMap.clear();
```

Comment:
Do we want to retain this if the attach fails, maybe some of the values would be the same across thread dumps without a change

Reply:
You are right. I made a change to reset all values if the attach fails so that it gets re-initialized as it would after a restart.